drivers/platform/stm32/stm32_i2c.c

```diff
@@ -95,8 +95,10 @@ int32_t stm32_i2c_init(struct no_os_i2c_desc **desc,
 	};
 
 	xdesc->hi2c.Instance = base;
+#if defined (STM32F4)
```

Comment:
how is the speed configured in STM32L5/H5 families? similar question for duty cycle

Reply:
There is an "I2C Speed Frequency (KHz) " option to be configured via the .ioc file.
I assume that the duty cycle is taken as a default 50% since it's a clock pulse?

Reply:
the speed should be configured in code based on the init parameter (we are overwriting the settings in the .ioc), we should be consistent  for all families...so we need to find the equivalent for ClockInit and configure it in the code for L5/H5

Reply:
@rbolboac - The speed is configured via the I2C timing register in some parts. The HAL driver does not provide separate bit masks for each bit in the timing register. Hence, I've added a value in the stm32_i2c_init_params so that the user can configure it according to the required frequency and duty ratio. Does this look ok?

Reply:
One thing I did not think about is this check for STM32F4... are these registers available only for this family?

Reply:
The timing register is not present in STM32F4. 
The F4 handles the frequency and duty ratio via the "I2C Clock Control Register"

Reply:
yes, but not only F4, from here https://github.com/search?q=org%3ASTMicroelectronics+hi2c-%3EInit.ClockSpeed&type=code&p=1 it looks like there a few other mcu families which use the same approach

Reply:
when building for STM32F1, STM32F2, STM32L1 the current implementation will probably lead to a compilation error. So I would suggest to also add these definitions checks besides STM32F4

Reply:
hopefully these are all....